bftengine/src/preprocessor/PreProcessor.hpp

```diff
@@ -75,11 +76,13 @@ class PreProcessor {
   void registerRequest(ClientPreProcessReqMsgUniquePtr clientReqMsg,
                        PreProcessRequestMsgSharedPtr preProcessRequestMsg);
   void releaseClientPreProcessRequestSafe(uint16_t clientId);
-  void releaseClientPreProcessRequest(ClientRequestInfoSharedPtr clientEntry, uint16_t clientId);
+  void releaseClientPreProcessRequest(const ClientRequestInfoSharedPtr &clientEntry, uint16_t clientId);
```

Comment:
Why not pass by value, as usual for shared ptrs?

Reply:
Though the use of shared pointers has a low overhead, in case we know that a pointer is not expected to be stored/reset/changed in the code it is passed to, better to pass it by const&.

Reply:
Thanks for the clarification.